service/prometheus/update.go

```diff
@@ -31,11 +31,18 @@ func UpdateConfig(promcfg config.Config, scrapeConfigs []config.ScrapeConfig) (c
 // isManaged returns true if the given scrape config is managed by the prometheus-config-controller,
 // false otherwise.
 func isManaged(scrapeConfig config.ScrapeConfig) bool {
+	// TODO: the static config detection can be removed once all installations use k8s sd.
```

Comment:
Tracked in https://github.com/giantswarm/giantswarm/issues/1482, we only need to keep it for the migration. I'm gonna do this next.